packages/office-ui-fabric-react/src/components/Link/Link.base.tsx

```diff
@@ -25,53 +18,28 @@ export class LinkBase extends BaseComponent<ILinkProps, any> implements ILink {
       theme: theme!
     });
 
-    const anchorNativeProps = getNativeProps(this.props, anchorProperties);
-    const buttonNativeProps = getNativeProps(this.props, buttonProperties);
+    const RootType = this._getRootType(this.props);
 
-    const anchorElement: JSX.Element = (
+    return (
       <KeytipData
         keytipProps={keytipProps}
-        ariaDescribedBy={(anchorNativeProps as any)['aria-describedby']}
+        ariaDescribedBy={(this.props as { 'aria-describedby': string })['aria-describedby']}
         disabled={disabled}
       >
         {(keytipAttributes: any): JSX.Element => (
-          <a
-            {...anchorNativeProps}
+          <RootType
             {...keytipAttributes}
+            {...this._removeInvalidPropsForRootType(RootType, this.props)}
             className={classNames.root}
             onClick={this._onClick}
             ref={this._link}
-            target={this.props.target}
             aria-disabled={disabled}
           >
             {children}
-          </a>
+          </RootType>
         )}
       </KeytipData>
     );
-
-    const buttonElement: JSX.Element = (
-      <KeytipData
-        keytipProps={keytipProps}
-        ariaDescribedBy={(buttonNativeProps as any)['aria-describedby']}
-        disabled={disabled}
-      >
-        {(keytipAttributes: any): JSX.Element => (
-          <button
-            {...buttonNativeProps}
-            {...keytipAttributes}
-            className={classNames.root}
-            onClick={this._onClick}
-            ref={this._link}
-            aria-disabled={disabled}
-          >
-            {children}
-          </button>
-        )}
-      </KeytipData>
-    );
-
-    return href ? anchorElement : buttonElement;
   }
 
   public focus() {
```

Comment:
I think there's a bug in focus now; since you can override as with anything, that could potentially be something that doesn't implement `ILink`, implying that you'd null ref on line 47. Perhaps we should check for the existence of the focus method?